src/core/inc/GslBlockMatrix.h

```diff
@@ -0,0 +1,94 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2015 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef UQ_GSL_BLOCK_MATRIX_H
+#define UQ_GSL_BLOCK_MATRIX_H
+
+/*!
+ * \file GslBlockMatrix.h
+ * \brief QUESO block matrix class using GSL.
+ */
+
+#include <vector>
+#include <queso/Environment.h>
+#include <queso/GslVector.h>
+#include <queso/GslMatrix.h>
+#include <queso/VectorSpace.h>
+
+namespace QUESO {
+
+/*!
+ * \class GslBlockMatrix
+ * \brief Class for representing block matrices using GSL library.
+ *
+ * This class provides basic 'invertMultiply' support for matrices of block
+ * diagonal structure.  Each block is implemented as a GslMatrix object.
+ */
+
+class GslBlockMatrix
```

Comment:
>That said, I wouldn't object to having the missing pure virtual methods throw a queso_not_implemented() so this can move forward.

Thanks for the input.  I've pushed a commit that addresses this.